lib/Doctrine/ODM/MongoDB/Configuration.php

```
@@ -180,17 +192,34 @@ public function getAutoGenerateProxyClasses() : int
      */
     public function setAutoGenerateProxyClasses(int $mode) : void
     {
-        $this->attributes['autoGenerateProxyClasses'] = $mode;
+        $this->autoGenerateProxyClasses = $mode;
+        $proxyManagerConfig             = $this->getProxyManagerConfiguration();
+
+        switch ($mode) {
+            case self::AUTOGENERATE_ALWAYS:
+            case self::AUTOGENERATE_FILE_NOT_EXISTS:
+                $proxyManagerConfig->setGeneratorStrategy(new FileWriterGeneratorStrategy(
+                    new FileLocator($proxyManagerConfig->getProxiesTargetDir())
+                ));
+
+                break;
+            case self::AUTOGENERATE_NEVER:
+            case self::AUTOGENERATE_EVAL:
+            default:
```

IMO `default` should throw `\InvalidArgumentException`

---

Yeah, makes sense! 👍